test/Unit/Runners/PHPUnit/WrapperRunnerTest.php

```diff
@@ -42,4 +42,13 @@ public function testWrapperRunnerWorksWellWithManyTests(): void
 
         $this->runRunner();
     }
+
+    public function testWrapperRunnerHandlesBatchSize(): void
+    {
+        $this->bareOptions['--path']           = $this->fixture('passing_tests' . DS . 'level1' . DS . 'level2');
+        $this->bareOptions['--configuration']  = $this->fixture('phpunit-parallel-suite-with-globals.xml');
+        $this->bareOptions['--max-batch-size'] = 2;
+
+        $this->runRunner();
```

Comment:
This is not a test, it's just a mere execution of something.

A proper test for the `maxBatchSite` should:

1. Consist of 4 separate tests
2. Run in 2 processes
3. Each test should track both `getmypid()` and `TEST_TOKEN` env var unique values
4. Assert that `getmypid()` unique values are `4`, and `TEST_TOKEN` unique values are `2`

Reply:
I didn't think about using the testcases that way.

I think I did come up with something useful